### PR TITLE
Support local config file

### PR DESCRIPTION
**Pull request by ooga**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -134,3 +134,14 @@ Run `gitmoji -g` to setup some gitmoji-cli preferences.
 - **Scope prompt**: Enable or disable [conventional commits scope prompt](https://www.conventionalcommits.org/en/v1.0.0/#summary).
 - **Signed commits**: Enable or disable [signed commits with GPG](https://docs.github.com/en/free-pro-team@latest/github/authenticating-to-github/signing-commits).
 - **Gitmojis api URL**: Set a custom URL to use it as the library of gitmojis.
+
+#### Configuration file
+
+A local `.prettierrc` file written in JSON can be added to your project to share your configuration
```
Comment (Addono) [inline README.md:140]:
```suggestion
A local `.gitmojirc` file written in JSON can be added to your project to share your configuration
```

File: README.md [hunk 2]
```diff
@@ -134,3 +134,14 @@ Run `gitmoji -g` to setup some gitmoji-cli preferences.
...
+
+Just add a `.prettierrc` with empty object
```
Comment (Addono) [inline README.md:142]:
```suggestion
Just add a `.gitmojirc` with empty object
```

File: src/utils/configurationVault.js [hunk 3]
```diff
@@ -1,12 +1,31 @@
...
+const confOptions = hasLocalConfig 
+  ? {
+    cwd: process.cwd(),
+    configName: localConfigFile,
+    fileExtension: undefined
+  }
+  : {
+    projectName: 'gitmoji'
+  };
+
```
Comment (carloscuesta) [inline src/utils/configurationVault.js:27]: I would set this into a function like:

```suggestion
const getConfiguration = () => {
  try {
    // We should validate in some way the configuration options obtained from the file/package.json
    if (fs.existsSync(GITMOJI_CONFIGURATION_OPTIONS)) {
      return {
        cwd: process.cwd(),
        configName: localConfigFile,
        fileExtension: undefined
      }
    }

    // Get configuration options from package.json if exist apply them

    return { projectName: 'gitmoji' }
  } catch (error) {
    
  }
}
```

File: src/utils/configurationVault.js [hunk 4]
```diff
@@ -1,12 +1,31 @@
...
+export const config = new Conf(confOptions)
```
Comment (carloscuesta) [inline src/utils/configurationVault.js:28]:
```suggestion
export const config = new Conf(getConfiguration())
```

**Review by carloscuesta — COMMENTED** [review]
Hey! Thanks for opening the PR, just added you a comment

It would be great to introduce the package.json as a source of configuration options to reduce the amount of .dotfiles!

**Comment by carloscuesta** [comment]
Hey @ooga 👋🏼 

Do you have the time to work on this PR? If not I will find some time and I will try to implement it

**Comment by ooga** [comment]
Hi @carloscuesta,

No, I'm not working on it.
What you suggest is quite challenging to integrate with the current `--config` cli option.
It will need to rewrite `configurationVault.js` to handle different source type of configuration

**Comment by carloscuesta** [comment]
Sounds good! 😊 

I'll close this then

Thanks 🙏🏼

**Pull request closed without merging**